src/widgets/helper/ChannelView.cpp

```diff
@@ -940,6 +957,41 @@ void ChannelView::messageReplaced(size_t index, MessagePtr &replacement)
     this->queueLayout();
 }
 
+void ChannelView::messagesUpdated()
+{
+    auto snapshot = this->channel_->getMessageSnapshot();
+
+    this->messages_.clear();
+    this->scrollBar_->clearHighlights();
+    this->lastMessageHasAlternateBackground_ = false;
+    this->lastMessageHasAlternateBackgroundReverse_ = true;
+
+    for (const auto &msg : snapshot)
+    {
+        auto messageLayout = new MessageLayout(msg);
```

Comment:
Why not:
```suggestion
        auto messageLayout = std:.make_shared<MessageLayout>(msg);
```
(And then moving the `shared_ptr` into `this->messages_.push_back()` later.)

I would feel safer that way

Reply:
I went ahead and made this change (and the change in other places in the file where similar code occurs). The only """issue""" is that `LimitedQueue` doesn't have a version of `pushBack` that takes an rvalue, so (I think) that means we can't `std::move` into the `LimitedQueue`. Not a big deal.

Reply:
Good catch, didn't even think about that. At least we're future-proof this way if anyone ever adds a rvalue version (which will surely happen ...)